src/games/Lexible/models/WordTree.ts

```diff
@@ -1,44 +1,257 @@
+    child(letter: string): WordTreeSearcher | undefined;
+}
+
+export interface WordTreeUsageRecord {
+    nodeCount: number;
+    filledChildPointers: number;
+    totalChildPointers: number;
+}
+
+class WordTreeUsageRecordImpl implements WordTreeUsageRecord {
+    nodeCount: number = 0;
+    filledChildPointers: number = 0;
+    totalChildPointers: number = 0;
+    
+    toString(): string {
+        return `WordTreeUsageRecord (${this.nodeCount} nodes, ${this.filledChildPointers}/${this.totalChildPointers} pointers filled)`
+    }
+}
+
+export interface WordTreeUsageStats {
+    nodeCount: number;
+    wordCount: number;
+    totalChildPointers: number;
+    filledChildPointers: number;
+    usageStatsByLetter: Map<string, WordTreeUsageRecord>
+    usageStatsByLength: Map<number, WordTreeUsageRecord>
+}
 
 // -------------------------------------------------------------------
 // A class for rapidly finding potential word matches
 // -------------------------------------------------------------------
 export class WordTree {
-    private _isTerminator = false;
-    private _branches = new Map<string, WordTree>()
-    private _letter: string
-    private _parent: WordTree | undefined
+    // TODO: Do bounds checks
+    private _buffer: ArrayBuffer;
+    private _allocationPoint: number;
+    private _totalNodeCount: number;
+    private _totalWordCount: number;
 
-    get rootWord():string { return (this._parent?.rootWord ?? "") + this._letter}
-    get myWord() { return this._isTerminator ?  this.rootWord: undefined}
+    constructor() {
+        this._buffer = new ArrayBuffer(RECORD_SIZE + FIRST_NODE_LOCATION);
+        this._allocationPoint = FIRST_NODE_LOCATION + RECORD_SIZE;
+        this._totalNodeCount = 1;
+        this._totalWordCount = 0;
+    }
 
-    static create(words: string[])
-    {
-        const output = new WordTree("", undefined);
-        words.forEach(w => output.add(w.toUpperCase()))
-        return output;
+    add(word: string) {
+        this.addInternal(word, FIRST_NODE_LOCATION);
+        this._totalWordCount++;
     }
 
-    constructor(letter: string, parent: WordTree | undefined){
-        this._letter = letter;
-        this._parent = parent;
+    search(): WordTreeSearcher {
+        return new WordTree.WordTreeSearcherRoot(this);
     }
 
-    branch(letter: string) {
-        return this._branches.get(letter)
+    trim(): void {
+        this.resize(this._allocationPoint);
     }
 
-    add(word: string) {
-        if(word.length === 0) {
-            this._isTerminator = true;
+    hasWord(word: string) {
+        word = word.toUpperCase();
+        let searcher: WordTreeSearcher = this.search();
+        for (const letter of word) {
+            const next = searcher.child(letter);
+            if (!next) return false;
+            searcher = next;
+        }
+        return searcher.isTerminator;
+    }
+
+    private static WordTreeSearcherImpl = class WordTreeSearcherImpl implements WordTreeSearcher {
```

Comment:
nit:  I prefer
```
private static WordTreeSearcher = class WordTreeSearcher implements IWordTreeSearcher
``` 
... because I feel like Impl tells me less than than the I* prefix.  e.g.:  If I see a class property `foo: IWordTreeSearcher`, it's clear that I can provide any kind of searcher there without having to check the type.  We would never use the *Impl version this way on a property.

I do support naming implementations to indicated the type of implementation.   e.g.:  MockWordTreeSearcher or SQLWordTreeSearcher

Just rattling off some thoughts.